qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/RecallIT.java

```diff
@@ -0,0 +1,83 @@
+    private static final String TEST_INDEX_RECALL_OLD = KNN_BWC_PREFIX + "test-index-recall-old";
+    private static final String TEST_FIELD = "test-field";
+    private static final double DELTA_VALUE = 0.2;
+    private static final int DIMENSIONS = 50;
+    private static final int DIMENSIONS_RECALL_OLD = 1;
+    private static final String DOC_ID = "1";
+    private static final int INDEX_THREAD_QTY = 2;
+    private static final int K = 10;
+    private static final int NUM_DOCS = 1000;
+    private static final int QUERY_COUNT = 100;
+
+    public void testKNNRecall() throws Exception {
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+        if (isRunningAgainstOldCluster()) {
+            createKnnIndex(testIndex, getKNNDefaultIndexSettings(), createKnnIndexMapping(TEST_FIELD, DIMENSIONS));
+            addDocs(testIndex, TEST_FIELD, DIMENSIONS, NUM_DOCS, true);
+
+            double recallVal = getKNNBWCRecallValue(testIndex, TEST_FIELD, NUM_DOCS, DIMENSIONS, QUERY_COUNT, K, true, SpaceType.L2);
+            createKnnIndex(TEST_INDEX_RECALL_OLD, getKNNDefaultIndexSettings(), createKnnIndexMapping(TEST_FIELD, DIMENSIONS_RECALL_OLD));
+            addKnnDoc(TEST_INDEX_RECALL_OLD, DOC_ID, TEST_FIELD, new Float[] { (float) recallVal });
+
+        } else {
+            double recallValUpgraded = getKNNBWCRecallValue(
+                testIndex,
+                TEST_FIELD,
+                NUM_DOCS,
+                DIMENSIONS,
+                QUERY_COUNT,
+                K,
+                true,
+                SpaceType.L2
+            );
+            float[][] expRecallValue = getIndexVectorsFromIndex(TEST_INDEX_RECALL_OLD, TEST_FIELD, 1, DIMENSIONS_RECALL_OLD);
+            assertEquals(expRecallValue[0][0], recallValUpgraded, DELTA_VALUE);
+
+            deleteKNNIndex(testIndex);
+            deleteKNNIndex(TEST_INDEX_RECALL_OLD);
+        }
+    }
+
+    private double getKNNBWCRecallValue(
+        String testIndex,
+        String testField,
+        int docCount,
+        int dimensions,
+        int queryCount,
+        int k,
+        boolean isStandard,
+        SpaceType spaceType
+    ) throws Exception {
+        float[][] indexVectors = getIndexVectorsFromIndex(testIndex, testField, docCount, dimensions);
+        float[][] queryVectors = TestUtils.getQueryVectors(queryCount, dimensions, docCount, isStandard);
+        List<Set<String>> groundTruthValues = TestUtils.computeGroundTruthValues(indexVectors, queryVectors, spaceType, k);
+        List<List<String>> searchResults = bulkSearch(testIndex, testField, queryVectors, k);
+        return TestUtils.calculateRecallValue(searchResults, groundTruthValues, k);
+    }
+
+    private void addDocs(String testIndex, String testField, int dimensions, int docCount, boolean isStandard) throws Exception {
+        updateClusterSettings(KNN_ALGO_PARAM_INDEX_THREAD_QTY, INDEX_THREAD_QTY);
+        updateClusterSettings(KNN_MEMORY_CIRCUIT_BREAKER_ENABLED, true);
```

Comment:
Why are we adding these 2 lines?

Reply:
We are updating index thread qty to 2 to speed up data ingestion. Also, when we are ingesting more documents, sometimes it is throwing heap out of memory exception. To avoid this, we are enabling circuit breaker.

Reply:
But KNN_MEMORY_CIRCUIT_BREAKER_ENABLED wont help avoid memory issues while indexing. If we are getting memory issues, I would probably reduce the number of documents to either 500 or 100.

Reply:
Got it. Updated